src/main/java/com/refinedmods/refinedstorage/util/SaveDataManager.java

```diff
@@ -0,0 +1,133 @@
+package com.refinedmods.refinedstorage.util;
+
+import com.refinedmods.refinedstorage.apiimpl.storage.disk.StorageDiskManager;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.CompressedStreamTools;
+import net.minecraft.util.RegistryKey;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraft.world.storage.FolderName;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class SaveDataManager {
+    public static final SaveDataManager INSTANCE = new SaveDataManager();
+    private static final Logger LOGGER = LogManager.getLogger(SaveDataManager.class);
+    private final Map<RegistryKey<World>, Map<Class<?>, ISaveData>> worldSaveData = new HashMap<>();
+    private final Map<Class<?>, Supplier<ISaveData>> managerTypes = new LinkedHashMap<>();
+
+    public void registerManager(Class<?> clazz, Supplier<ISaveData> supplier) {
+        managerTypes.put(clazz, supplier);
+    }
+
+    private void createManagers(RegistryKey<World> worldKey) {
+        Map<Class<?>, ISaveData> map = new LinkedHashMap<>();
+        managerTypes.forEach((clazz, supplier) -> {
+            if (clazz == StorageDiskManager.class) {
+                if (worldKey == World.field_234918_g_) {
+                    map.put(clazz, supplier.get());
+                }
+            } else {
+                map.put(clazz, supplier.get());
+            }
+        });
+        worldSaveData.put(worldKey, map);
+    }
+
+    public <T extends ISaveData> T getManager(Class<T> clazz, RegistryKey<World> worldKey) {
+        return clazz.cast(worldSaveData.get(worldKey).get(clazz));
+    }
+
+    public void removeManagers(ServerWorld world) {
+        worldSaveData.remove(world.func_234923_W_());
+    }
+
+    public void read(ServerWorld world) {
+        RegistryKey<World> worldKey = world.func_234923_W_();
+        if (!worldSaveData.containsKey(worldKey)) {
+            createManagers(worldKey);
+        }
+
+        for (ISaveData saveDatum : worldSaveData.get(worldKey).values()) {
+            CompoundNBT nbt = new CompoundNBT();
+            try {
+                nbt = readTagFromFile(world, saveDatum.getName());
+            } catch (IOException e) {
+                LOGGER.warn("Unable to read " + saveDatum.getName() + "'s backup. Continuing without data");
```

Comment:
Can't you pass the context of the exception to LOGGER.warn instead of printStackTrace?

Reply:
I could not find it when I looked last.. But yep I can.